scheduler/core/service/service.py

```diff
@@ -15,18 +15,28 @@
 
 
 class Service:
-    def __init__(self, start_time: Time, end_time: Time, sites: FrozenSet[Site]):
+    def __init__(self,
+                 start_time: Time,
+                 end_time: Time,
+                 semesters: FrozenSet[Semester],
```

Comment:
Why are the semesters now part of the `Service` class?
I'm not clear why we're moving this data around.